commands/base_commands/social.py

```diff
@@ -2191,7 +2192,28 @@ def func(self):
         """Execute command."""
         caller = self.caller
         if not self.switches or self.check_switches(self.prestige_switches):
-            return self.get_queryset_for_prestige_table()
+            if not self.lhs:
+                return self.get_queryset_for_prestige_table()
+            else:
+                targ = caller.search(self.lhs)
```

Comment:
This is nice! If you wanted to, you could make it generic to org or player by searching for the AssetOwner directly, and searching either via the org name or `player__player__username`. Longname would only be applicable in the character case though.

Reply:
I think organisations should probably get longnames too. Right now the Knights of Solace are just 'Solace' while the Faith are 'Faith of the Pantheon', it's very inconsistent.

Reply:
We have long names for player characters because there's many situations where you want to only have their name be briefly mentioned rather than their full name. Does the same really apply to organizations? I guess in tables, maybe. It'd probably make more sense to have add a short name as a SlugField for better urls for org pages and use it in tables, I guess, and then use the current name as the verbose/full name.

Reply:
I know I keep trying to go 'org faith' and 'org explorers' and the game saying mean things to me about not being a member of such an org.

Reply:
For command matching we just need to update some of those commands to use [this method](https://github.com/Arx-Game/arxcode/blob/stable_orphan/commands/mixins.py#L52) that @Tanookami wrote. It does a contains check first for strings, then tries to narrow it down on multimatches with iexact.

Reply:
Unless I'm missing something, I can't search for the AssetOwner directly because AssetOwners don't have names, I'd have to grab all the Assets, then grab all the owners, then filter the owners.

What I could do is first try to find a playerornpc with the name, then try to find an org with the name, then grab their assets.

Reply:
```
qs = AssetOwner.objects.filter(Q(player__player__username__icontains=self.lhs) | Q(organization_owner__name__icontains=self.lhs))
if len(qs) > 1:
    qs = qs.filter(Q(player__player__username__iexact=self.lhs) | Q(organization_owner__name__iexact=self.lhs))
if len(qs) > 1:
    self.msg("Please narrow results.")
    return
obj = qs.first()
if not obj:
    self.msg("Nothing found")
    return
```
Something like that, probably.